primitives/storage/src/lib.rs

```diff
@@ -98,6 +98,11 @@ pub mod well_known_keys {
 	/// Prefix of child storage keys.
 	pub const CHILD_STORAGE_KEY_PREFIX: &'static [u8] = b":child_storage:";
 
+	/// The last time the runtime upgrade happened.
+	///
+	/// Stores the the `impl_version` and the `spec_version` of this runtime.
+	pub const LAST_RUNTIME_UPGRADE: &'static [u8] = b":last_runtime_upgrade:";
```

Comment:
This is a bit FRAME specific. Other runtimes might want to employ other techniques for detecting a runtime upgrade and they also might not have the notion of `impl_version` and `spec_version`. Probably it would be better to move somewhere under FRAME.